build.gradle

```diff
@@ -73,6 +83,22 @@ subprojects {
   apply plugin: 'nebula.dependency-recommender'
   apply plugin: 'java-library'
 
+  // apply this plugin in a try-catch block so that we can handle cases without .git directory
+  // apply to subprojects so that git properties information is available for runtime debugging and tests
+  //  e.g. sending build SHA as headers
+  try {
+    apply plugin: 'com.gorylenko.gradle-git-properties'
```

Comment:
We only need one file, `build/iceberg-build.properties`. Can you remove the others?

Reply:
From my testing, we would need it for local development or using a JAR built locally (such as copying over one `iceberg-core.jar` or even the spark bundle file).

But I can remove this.

Reply:
This has been removed. Note that from my previous testing, the file in `build/iceberg-build.properties` will not be available on the classpath when running in the IDE or during gradle tests. We can resolve that in subsequent PRs if need be though.

Also per the stack overflow that was the basis of https://github.com/apache/iceberg/pull/5186